tests/integration/baseTestValidator.py

```diff
@@ -116,57 +116,58 @@ def run_test(self, jobId, statusId, statusName, fileSize, stagingRows,
-        self.assertEqual(response.status_code, statusId,
-            msg="{}".format(self.getResponseInfo(response)))
-        if statusName != False:
-            self.waitOnJob(jobTracker, jobId, statusName, self.useThreads)
-            self.assertEqual(jobTracker.getJobStatus(jobId), jobTracker.getJobStatusId(statusName))
-
-        self.assertEqual(
-            response.headers.get("Content-Type"), "application/json")
-
-        # Check valid row count for this job
-        if stagingRows:
-            numValidRows = jobTracker.getNumberOfValidRowsById(jobId)
-            self.assertEqual(numValidRows, stagingRows)
-
-        errorInterface = self.errorInterface
-        if errorStatus is not False:
-            self.assertEqual(errorInterface.checkFileStatusByJobId(jobId), errorInterface.getFileStatusId(errorStatus))
-            self.assertEqual(errorInterface.checkNumberOfErrorsByJobId(jobId, self.validationDb,"fatal"), numErrors)
-            self.assertEqual(errorInterface.checkNumberOfErrorsByJobId(jobId, self.validationDb,"warning"), numWarnings)
-        if(fileSize != False):
-            if self.local:
-                self.assertFileSizeAppxy(
-                    fileSize, jobTracker.getReportPath(jobId))
-            else:
-                self.assertGreater(s3UrlHandler.getFileSize(
-                    "errors/"+jobTracker.getReportPath(jobId)), fileSize - 5)
-                self.assertLess(s3UrlHandler.getFileSize(
-                    "errors/"+jobTracker.getReportPath(jobId)), fileSize + 5)
-        if(warningFileSize is not None and warningFileSize != False):
-            if self.local:
-                self.assertFileSizeAppxy(
-                    warningFileSize, jobTracker.getWarningReportPath(jobId))
-            else:
-                self.assertGreater(s3UrlHandler.getFileSize(
-                    "errors/"+jobTracker.getWarningReportPath(jobId)), warningFileSize - 5)
-                self.assertLess(s3UrlHandler.getFileSize(
-                    "errors/"+jobTracker.getWarningReportPath(jobId)), warningFileSize + 5)
+        with createApp().app_context():
+            sess = GlobalDB.db().session
+
+            response = self.validateJob(jobId)
+            self.assertEqual(response.status_code, statusId, msg="{}".format(self.getResponseInfo(response)))
+
+            # get the job from db
+            job = sess.query(Job).filter(Job.job_id == jobId).one()
+            if statusName is not False:
+                self.assertEqual(job.job_status_id, JOB_STATUS_DICT[statusName])
+
+            self.assertEqual(
+                response.headers.get("Content-Type"), "application/json")
+
+            # Check valid row count for this job
+            if stagingRows:
+                self.assertEqual(job.number_of_rows_valid, stagingRows)
+
+            if errorStatus is not False:
+                self.assertEqual(
+                    sess.query(File).filter(File.job_id == jobId).one().file_status_id,
```

Comment:
The point re: legibility is fair, but since the Job model doesn't have a relationship back to `file` (only `file_type`), I think I'm going to leave this for now and not pull the thread of re-jiggering the Job model.